### PR TITLE
MOSIP-20457 Added Junit for BaseWebSubInitializer.

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

**Review by loganathan-sekaran — APPROVED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java [hunk 1]
```diff
@@ -124,7 +117,9 @@ private void subscribe(String topic, String callbackUrl, String secret, String h
             subscriptionRequest.setSecret(secret);
             subscriptionRequest.setTopic(topic);
             subscriptionRequest.setHubURL(hubUrl);
-            subscribe.subscribe(subscriptionRequest);
+            if (subscribe != null) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: This check is not needed.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java [hunk 2]
```diff
@@ -95,26 +93,21 @@ protected void tryRegisterTopicEvent(String eventTopic) {
...
-                            + callbackUrl);
-            subscribe(topic, callbackUrl, secret, hubUrl);
-            logger.info("subscribe", "",
-                    "Subscribed to topic: " + topic);
-        } catch (Exception e) {
-            logger.info("subscribe", e.getClass().toString(),
-                    "Error subscribing topic: " + topic + "\n" + e.getMessage());
-            throw e;
-        }
+    public void initSubsriptions() {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: Make it to protected

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java [hunk 3]
```diff
@@ -95,26 +93,21 @@ protected void tryRegisterTopicEvent(String eventTopic) {
...
+                "Trying to subscribe to topic: " + topic + " callback-url: "
+                        + callbackUrl);
+        subscribe(topic, callbackUrl, secret, hubUrl);
+        logger.info("subscribe", "",
+                "Subscribed to topic: " + topic);
+
     }
 
-    private void subscribe(String topic, String callbackUrl, String secret, String hubUrl) {
+    public void subscribe(String topic, String callbackUrl, String secret, String hubUrl) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: Make it as private

File: resident/resident-service/src/test/java/io/mosip/resident/test/service/BaseWebSubInitializerTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.cloud.context.config.annotation.RefreshScope;
+import org.springframework.http.HttpHeaders;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.util.ReflectionTestUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+@RefreshScope
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/test/java/io/mosip/resident/test/service/BaseWebSubInitializerTest.java]: Do we need @RefreshScope? Please remove it and check.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java (hunk 1 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: Removed this check.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java (hunk 2 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: Done.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java (hunk 3 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: Done.

File: resident/resident-service/src/test/java/io/mosip/resident/test/service/BaseWebSubInitializerTest.java (hunk 4 above)
Comment (kameshsr) [reply resident/resident-service/src/test/java/io/mosip/resident/test/service/BaseWebSubInitializerTest.java]: Removed @RefreshScope. it's working.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java [hunk 5]
```diff
@@ -77,16 +80,11 @@ public void onApplicationEvent(ApplicationReadyEvent applicationReadyEvent) {
...
-        }
-        catch (Exception e){
-            logger.error( "authTransactionSubscription", "BaseWebSubInitializer", "Exception while subscribing to topic: " + e.getMessage());
-        }
+    public void authTransactionSubscription() {
+        subscribe(authTransactionTopic, callbackAuthTransactionUrl, authTransactionSecret, hubUrl);
     }
 
-    protected void tryRegisterTopicEvent(String eventTopic) {
+    public void tryRegisterTopicEvent(String eventTopic) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: Do you need public modifier here? can it remain protected?

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java (hunk 5 above)
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/BaseWebSubInitializer.java]: No changed it to protected.

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**